components/lib/state/src/main/java/mozilla/components/lib/state/Store.kt

```diff
@@ -39,9 +40,9 @@ typealias Observer<S> = (S) -> Unit
  */
 open class Store<S : State, A : Action>(
     initialState: S,
-    private val reducer: Reducer<S, A>
+    private val reducer: Reducer<S, A>,
+    dispatcher: CoroutineDispatcher = Executors.newSingleThreadExecutor().asCoroutineDispatcher()
```

Comment:
We shouldn't need expose our dispatcher. I am a bit worried what client-code could do with this :). It also shouldn't be necessary for tests.